frontend/pages/Homepage/ActivityFeed/ActivityFeed.tsx

```diff
@@ -0,0 +1,174 @@
+import React, { useEffect } from "react";
+// @ts-ignore
+import Fleet from "fleet";
+import { create, lowerCase } from "lodash";
+
+import Avatar from "components/Avatar";
+
+const baseClass = "activity-feed";
+
+const activities = [
+  {
+    id: 1,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "created_pack",
+    details: {
+      pack_id: 1,
+      pack_name: "Cool pack",
+    },
+  },
+  {
+    id: 2,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "deleted_pack",
+    details: {
+      pack_name: "Cool pack",
+    },
+  },
+  {
+    id: 3,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "edited_pack",
+    details: {
+      pack_id: 1,
+      pack_name: "Cool pack",
+    },
+  },
+  {
+    id: 4,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "created_saved_query",
+    details: {
+      query_id: 1,
+      query_name: "Awesome query",
+    },
+  },
+  {
+    id: 5,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "deleted_saved_query",
+    details: {
+      query_name: "Awesome query",
+    },
+  },
+  {
+    id: 6,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "edited_saved_query",
+    details: {
+      query_id: 1,
+      query_name: "Awesome query",
+    },
+  },
+  {
+    id: 7,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "created_team",
+    details: {
+      team_id: 1,
+      team_name: "Walmart Pay",
+    },
+  },
+  {
+    id: 8,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "deleted_team",
+    details: {
+      team_name: "Walmart Pay",
+    },
+  },
+  {
+    id: 9,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "live_query",
+    details: {
+      targets_count: 12030,
+    },
+  },
+  {
+    id: 10,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "applied_pack_spec",
+  },
+  {
+    id: 11,
+    created_at: "2020-11-05T05:09:44Z",
+    actor_full_name: "Jane Doe",
+    actor_id: 1,
+    type: "applied_query_spec",
+  },
+];
+
+interface IActivity {
+  id: number;
+  created_at: string;
+  actor_full_name: string;
+  actor_id: number;
+  type: string;
+  details?: {
+    [key: string]: any;
+  };
+}
+
+const ActivityFeed = (): JSX.Element => {
+  useEffect(() => {
+    const getActivities = async () => {
+      const a = await Fleet.activities.loadNext();
```

Comment:
This is great! After going through all the bits and pieces needed to mimic how we are currently handling the other APIs with Rachel, this is bliss.

Do you have any thoughts on defining hooks for GET/POST/etc, and in a component like this one just do something like:

```
const { loading, error, data } = useGetRequest('/v1/fleet/activities');
```

?

So any new API doesn't need any definition in the Fleet object, you just use it. The call itself can be implemented asynchronously so that we render loading states.

From a quick look into `Fleet` and `Base`, everything that is used to make the request can be stateless-ly gathered. If you don't use this API anywhere else, it's all pretty self contained and you can mock the hook in tests to test the different scenarios.

For data modifying you could define `post`, `patch`, etc pure functions, and it would work similarly. Or you could even do `const { updateSomething } = usePostRequest('/v1/fleet/something')` if you like `useMutation` type patterns.

Reply:
There's this https://www.npmjs.com/package/use-get-request-hook or a simple implementation: https://www.npmjs.com/package/use-get-request-hook